docs/NNNN-direct-clean-path-layout.md

```diff
@@ -0,0 +1,255 @@
+be sure, that no object identifier is  contained in another identifier as a 
+prefix path.  
+I.e. https://hdl.handle.net/XXXXX/test and https://hdl.handle.net/XXXXX/test/blah 
+could result to into storage root folders like  
+https/hdl.handle.net/XXXXX/test    
+https/hdl.handle.net/XXXXX/test/blah  
+This is an invalid OCFL storage root hierarchy.
+
+#### Length
+Several filesystems (e.g. Ext2/3/4) have byte restrictions on filename length.
+When using UTF16 (i.e. NTFS) or UTF32 characters in the filesystem, the byte length is double or quad of the character length.
+Using UTF8 characters in filesystems, byte length can only be determined by checking each string for maximum.  
+**Hint:** UTF8 has always less or equal bytes than UTF32 which means, that 
+assuming UTF32 instead of UTF8 for length calculation is safe, but would give you
+only 63 characters on 255 byte restrictions. 
+
+## Parameters
+
+### Summary
+
+* **Name:** `encodeUTF`
+    * **Description:** decides whether "dangerous" characters will be replaced by a 
+      defined replacement string or it's utf code (e.g. =u0020 for blank char) 
+    * **Type:** bool
+    * **Default:** false
+* **Name:** `maxFilenameLen`
+   * **Description:** determines the maximum number of characters within parts of the full pathname separated by `/` (files or folders).   
+     A result with more characters will raise an error.  
+    * **Type:** number
+   * **Constraints:** An integer greater 0
+   * **Default:** 127
+* **Name:** `maxPathnameLen`
+    * **Description:** determines the maximum number of result characters.
+      A result with more characters will raise an error.  
+    * **Type:** number
+    * **Constraints:** An integer greater 0
+    * **Default:** 32000
+* **Name:** `replacementString`
+  * **Description: String which is used to replace non-whitespace characters 
+    which need replacement.  
+    If `encodeUTF == true` only used for replacement of non-UTF8 characters.
+  * **Type:** string
+  * **Default:** "_"
+* **Name:** `whitespaceReplacementString`
+    * **Description:** String which is used to replace whitespaces (https://en.wikipedia.org/wiki/Template:Whitespace_(Unicode)).  
+      Only used if `encodeUTF == false`.  
+      **Hint:** if you want to remove (inner) whitespaces, just use the empty string
+    * **Type:** string
+    * **Default:** " " (U+0020)
+  
+## Procedure
+
+The following is an outline to the steps for mapping an identifier/filepath:
+
+### `utfEncode == true`
+
+
+1. Replace all non-UTF8 characters with `replacementString`
+2. Split the string at path separator "/"
+3. For each part do the following
+   1. Replace any character from this list with it's utf code in the form "=uXXXX" where XXXX is the code: U+0000-U+001f U+007f U+0020 U+0085 U+00a0 U+1680 U+2000-U+20a0 U+2028 U+2029 U+202f U+205f U+3000 \n \t * ? : [ ] " < > | ( ) { } & ' ! ; # @  
+   3. If part contains only periods Replace first period ("."), with UTF Code (U+002E)
```

Comment:
1. I would style "." as `.` and "U+002E" as `U+002E`. Is that the literal value that should be inserted or should it be `=u002E`?
2. I think this would read better as "If part only contains periods, replace..."